dev-java/protobuf-java/protobuf-java-3.20.1.ebuild

```diff
@@ -0,0 +1,92 @@
+# Copyright 2008-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# Tests not enabled, depend on com.google.truth which is not packaged
+# https://github.com/protocolbuffers/protobuf/blob/v21.4/java/core/pom.xml#L35-L40
+JAVA_PKG_IUSE="doc source"
+MAVEN_ID="com.google.protobuf:${PN}:${PV}"
+
+inherit java-pkg-2 java-pkg-simple
+
+PARENT_PN="${PN/-java/}"
+PARENT_PV="${PV}"
+PARENT_P="${PARENT_PN}-${PARENT_PV}"
+
+if [[ "${PV}" == "9999" ]]; then
+	inherit git-r3
+
+	EGIT_REPO_URI="https://github.com/protocolbuffers/${PARENT_PN}"
+	EGIT_SUBMODULES=()
+	EGIT_CHECKOUT_DIR="${WORKDIR}/${PARENT_P}"
+else
+	inherit verify-sig
+	SRC_URI="
+		https://github.com/protocolbuffers/${PARENT_PN}/archive/v${PARENT_PV}.tar.gz
+			-> ${PARENT_P}.tar.gz
+	"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86 ~amd64-linux ~x86-linux ~x64-macos"
+fi
+
+DESCRIPTION="Google's Protocol Buffers - Java bindings"
+HOMEPAGE="
+	https://developers.google.com/protocol-buffers/
+	https://github.com/protocolbuffers/protobuf
+"
+
+LICENSE="BSD"
+SLOT="0/31"
+
+S="${WORKDIR}/${PARENT_P}/java"
+
+BDEPEND="
+	dev-libs/${PARENT_PN}:${SLOT}
```

Comment:
That's very unfriendly to `qgrep dev-libs/protobuf`

Reply:
Hello @vaukai,

> That's very unfriendly to qgrep dev-libs/protobuf

I didn’t think of this case, you are right, I was overusing variables in a case it is now obviously not necessary.
Updated to use plain parent.

> And, reading [https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Subslots](https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Subslots#Non-binding_slot_dependencies) and given we're calling the executable I'd say it's the dev-foo/libfoo:* case.

I am a bit unsure about this, even if you are right I don’t know Google’s protobuf to be sure subpackage (Java, Ruby, Python) can use a non matching slot version for the parent package.
I prefer to let the maintainer taking the decision.

Reply:
> I didn’t think of this case, you are right, I was overusing variables in a case it is now obviously not necessary.
> [...]

Also in SRC_URI, see https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#Plaintexting_.24.7BPN.7D_in_SRC_URI

Reply:
Done applying variables for grep purposes in link paths.

Reply:
> Done applying variables for grep purposes in link paths.

But you are cluttering pgo with "alternative hompage" which is absolutely identical to remote-id